### DISCUSSION TITLE
Add a setting to allow UI changes to provisioned objects

### DISCUSSION THREAD

**Discussion by arash-bizcover** [grafana/grafana#57136] [Alerting]
We need to be able to change/modify objects (including alerts) that are provisioned by API with the UI. 

At the moment UI doesn't allow changes to the API provisioned objects. This kind of enforcement should have been a setting in the config so we could opt out of it if needed.
![image](https://user-images.githubusercontent.com/62287461/196297704-86c62cc2-08f2-43a2-975d-d70af9219687.png)

We need this as we have a bi-directional sync on UI and Code.

**Comment by arash-bizcover** [comment]
https://github.com/grafana/grafana/issues/57315

**Comment by ali-avci** [comment]
this enhancement might be accomplished in two steps, a delete might be easier to implement. Missed the delete feature during the development phase of the API code a lot.

**Comment by armandgrillet** [comment]
We have planned to do this for 9.3: https://github.com/grafana/grafana/issues/57911.

**Comment by MiggL** [comment]
Are there plans to implement this setting for file provisioning as well?

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
